src/audio/host-zephyr.c

```diff
@@ -433,6 +433,29 @@ static uint32_t host_get_copy_bytes_normal(struct host_data *hd, struct comp_dev
 	return ALIGN_DOWN(dma_copy_bytes, hd->dma_copy_align);
 }
 
+#if CONFIG_HOST_DMA_STREAM_SYNCHRONIZATION
+static inline bool stream_sync(struct host_data *hd, struct comp_dev *dev)
+{
+	if (!hd->is_grouped)
+		return true;
+
+	uint64_t current_time = k_cycle_get_64();
+
+	if (current_time >= hd->next_sync) {
```

Comment:
Having timer logic without wrap-around handling is always a bit iffy, but given we have explicit 64bit counter, it would seem this is safe. Even with a 10x faster system clock, this would still take thousands of years to wrap around. I think current uptime record holder is still the one Cisco switch that is around 26 years now, so we have some safety margin -> https://en.wikipedia.org/wiki/Uptime

Reply:
ack, worth an inline comment here so that no one tries to fix later on.